java-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/SecureYotiClient.java

```diff
@@ -127,11 +127,15 @@ private void validateReceipt(Receipt receipt) throws ActivityFailureException {
     }
 
     private Profile createProfile(byte[] profileBytes, Key secretKey) throws ProfileException {
-        EncryptedData encryptedData = parseProfileContent(profileBytes);
-        byte[] profileData = decrypt(encryptedData.getCipherText(), secretKey, encryptedData.getIv());
-        Map<String, Object> attributeMap = parseProfile(profileData);
-        Profile profile = createProfile(attributeMap);
-        return profile;
+        Map<String, Object> attributeMap = null;
+        if (profileBytes != null && profileBytes.length > 0) {
+            EncryptedData encryptedData = parseProfileContent(profileBytes);
+            byte[] profileData = decrypt(encryptedData.getCipherText(), secretKey, encryptedData.getIv());
+            attributeMap = parseProfile(profileData);
+        } else {
+            attributeMap = new HashMap<String, Object>();
```

Comment:
this code path could be eliminated by assigning an empty map in line 130 (https://github.com/getyoti/java/pull/3/files#diff-c6a1017541fac6236b06dbb075f182b0R130)